app/lib/account/backend.dart

```diff
@@ -31,6 +32,25 @@ void registerAccountBackend(AccountBackend backend) =>
 AccountBackend get accountBackend =>
     ss.lookup(#_accountBackend) as AccountBackend;
 
+/// Sets the active authenticated user.
+void registerAuthenticatedUser(AuthenticatedUser user) =>
+    ss.register(#_authenticated_user, user);
+
+/// The active authenticated user.
+AuthenticatedUser get authenticatedUser =>
+    ss.lookup(#_authenticated_user) as AuthenticatedUser;
+
+/// Calls [fn] with the currently authenticated user as an argument.
+///
+/// If no user is currently authenticated, this will throw an
+/// `UnauthorizedAccess` exception.
+Future<R> withAuthenticatedUser<R>(Future<R> fn(AuthenticatedUser user)) async {
+  if (authenticatedUser == null) {
+    throw UnauthorizedAccessException('No active user.');
+  }
+  return await fn(authenticatedUser);
```

Comment:
This only only works because `appengine` creates a _service scope_ per request... We are **not** going to clean this up here.. but I would really love to see this being less fragile. This basically and other reason why I want to get away from the `pub_server` abstractions.

Reply:
Yeah, and the tests doesn't help much either, because there is no guaranteed separation between the tests for accessing these fields. We have many reasons to refactor that part of the code :)